lib/runner/folder-walk.js

```diff
@@ -129,25 +129,27 @@ class Walker {
   }
 
   async applyTagFilter(list) {
-    if (!this.tags.anyTagsDefined() || this.usingCucumber) {
-      return list;
-    }
-
-    const matches = await Promise.all(list.map(filePath => {
-      return this.tags.loadModule(filePath);
-    }));
-
-    return matches.filter(context => {
-      if (!context) {
-        return false;
+    if (Array.isArray(list)) {
+      if (!this.tags.anyTagsDefined() || this.usingCucumber) {
+        return list;
```

Comment:
Instead of putting everything in the if block, we should return if `list` is not an array, and leave everything else as it is.

Reply:
I have made the changes sir